### PR TITLE
Make autoreject faster

### PR DESCRIPTION
**Pull request by jasmainak**
closes https://github.com/autoreject/autoreject/issues/35

@ramonapariciog can you try this branch and see if it's any faster?

### CODE REVIEW

File: autoreject/autoreject.py [hunk 1]
```diff
@@ -304,15 +304,21 @@ def func(thresh):
                 cache.update({thresh: obj})
             return cache[thresh]
 
-        initial_x = all_threshes[::5]
+        n_epochs = all_threshes.shape[0]
+        if n_epochs < 200:
+            idx = np.arange(0, n_epochs, 5)
+        else:
+            idx = np.linspace(0, n_epochs, 40, endpoint=False, dtype=int)
```
Comment (agramfort) [inline autoreject/autoreject.py]: to me the branching is not necessary. you get 40 points all the time approx

Comment (jasmainak) [inline autoreject/autoreject.py]: well if you have less than 40 epochs, then what do you do?

**Comment by agramfort** [comment]
then if n_epochs < 40. should be used?

**Comment by jasmainak** [comment]
> then if n_epochs < 40. should be used?

In the end, I did `np.unique(np.linspace(...))` ... to make sure it's always 40 points or less

**Comment by jasmainak** [comment]
@agramfort how does the PR look to you?

File: autoreject/autoreject.py [hunk 2]
```diff
@@ -366,6 +374,16 @@ def compute_thresholds(epochs, method='bayesian_optimization',
...
     ch_names = epochs_interp.ch_names
+
+    if n_jobs > 1:
+        my_thresh = delayed(_compute_thresh)
+        verbose = 10 if verbose is not False else 0
+        threshes = Parallel(n_jobs=n_jobs, verbose=verbose)(
+            my_thresh(data[:, pick], cv=cv, method=method,
+                      random_state=random_state) for pick in picks)
+        threshes = {ch_names[p]: thresh for p, thresh in zip(picks, threshes)}
+        return threshes
```
Comment (jasmainak) [inline autoreject/autoreject.py]: okay now we have a single code path for joblib.

**Comment by agramfort** [comment]
circle complains but diff LGTM

**Comment by jasmainak** [comment]
Circle fixed and PR merged. Thanks @agramfort for the review

**Pull request merged**